### ISSUE TITLE
[deferred-components]Flutter Module don't support deferred-components

### ISSUE THREAD

**Issue by canluhuang** [flutter/flutter#82791]
In the Flutter Application project, I successfully used the new feature of deferred-components. But in the Flutter Module project, the following error appears when compiling：

`  build/android_deferred_components_setup_files/plugin/build.gradle (created)
  build/android_deferred_components_setup_files/plugin/src/main/AndroidManifest.xml (created)
=================================================================================
                     Deferred components prebuild validation
=================================================================================
Newly generated android files:
  - deferred_components_module/build/android_deferred_components_setup_files/plugin/build.gradle
  - deferred_components_module/build/android_deferred_components_setup_files/plugin/src/main/AndroidManifest.xml
  - deferred_components_module/build/android_deferred_components_setup_files/app/src/main/res/values/strings.xml

The above files have been placed into `build/android_deferred_components_setup_files`,
a temporary directory. The files should be reviewed and moved into the project's
`android` directory.
---------------------------------------------------------------------------------

Setup verification can be skipped by passing the `--no-validate-deferred-components`
flag, however, doing so may put your app at risk of not functioning even if the
build is successful.
=================================================================================
Setup for deferred components incomplete. See recommended actions.
`
It seems that Flutter Module does not support deferred-components, and there are plans to support it in the future？

This's my version of Flutter: 
`[✓] Flutter (Channel beta, 2.2.0-10.3.pre, on Mac OS X 10.13.6 17G14042 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.2.0-10.3.pre at /Users/qq/Downloads/flutter_macos_2.2.0-10.3.pre-beta
    • Framework revision 06e2fd6357 (10 days ago), 2021-05-08 11:28:22 -0700
    • Engine revision a123e75c60
    • Dart version 2.13.0 (build 2.13.0-211.14.beta)
`

**Comment by chinmaygarde** [comment]
cc @GaryQian

**Comment by xster** [comment]
This is an interesting ask. We indeed didn't build support for this initially. How would you imagine this working? Your host application may itself have Android Feature Delivery configurations. How would the Flutter module signal its desired slicing? How would that work with gradle subprojects or AARs?

**Comment by xster** [comment]
Just to clarify, the way I understood this is that it's a different request from https://github.com/flutter/flutter/issues/81265. You don't want all of your Flutter module code along with the Flutter engine to be defer loaded. You want the main part of the Flutter module to be in the base APK but then defer other parts of your Flutter module to be loaded later on correct?

**Comment by wangtb** [comment]
@xster 

> Just to clarify, the way I understood this is that it's a different request from #81265. You don't want all of your Flutter module code along with the Flutter engine to be defer loaded. You want the main part of the Flutter module to be in the base APK but then defer other parts of your Flutter module to be loaded later on correct?

Is it possible? We do have this use case.

**Comment by GaryQian** [comment]
Yes, that use case is possible. In the deferred components config, you can specify which module is allocated to base apk and which is deferred. However, the libflutter.so framework dart code will always be included in the base apk though.

**Comment by wangtb** [comment]
Would you please provide some code sample, about how to " specify which
module is allocated to base apk and which is deferred"?
I can't find such doc from official website.
Thanks a lot!
